app/src/main/java/com/andrew/tasky/agenda/presentation/screens/task_detail/TaskDetailViewModel.kt

```diff
@@ -21,7 +23,8 @@ import kotlinx.coroutines.withContext
 @HiltViewModel
 class TaskDetailViewModel @Inject constructor(
     private val savedStateHandle: SavedStateHandle,
-    private val repository: AgendaItemRepository
+    private val localRepository: AgendaItemRepository,
+    private val remoteRepository: AgendaApiRepository
```

Comment:
The purpose of a repository is to combine multiple data sources, so the caching would be part of it, which is why you wouldn't need to distinguish between local and remote repo